File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/factory_method/AddFactory.java
package jv.com.tencent.cain.design_pattern.factory_method;

/**
 * @author cainjiang
 * @date 2019-05-20
 */
public class AddFactory implements IOperationFactory {
    @Override
    public Operation createOpration() {
        AddOperation addOperation = new AddOperation();
        addOperation.operationSymbol = "+";
        addOperation.operationName = "加法";
        return addOperation;
    }

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/factory_method/AddFactory.kt
package kt.com.tencent.cain.desigin_pattern.factory_method

import kt.com.tencent.cain.desigin_pattern.simple_factoty.AddOperation
import kt.com.tencent.cain.desigin_pattern.simple_factoty.Operation

/**
 * @author cainjiang
 * @date 2019-05-21
 */
class AddFactory : IOperationFactory {
    override fun createOpration(): Operation {
        val addOperation = AddOperation()
        addOperation.operationSymbol = "+"
        addOperation.operationName = "加法"
        return addOperation
    }
}